### PR TITLE
Update style, searchbar, and dropdown

### PR DESCRIPTION
**Pull request by BlackBeltPanda**
Update SettingsGui to new style and add new colors to VigilancePalette and VigilanceConfig
Deprecate old DropDown and replace with new DropDownComponent
Replace searchbar in SettingsTitle

### CODE REVIEW

File: src/main/kotlin/gg/essential/vigilance/gui/settings/DropDownComponent.kt [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+import gg.essential.vigilance.utils.hoveredState
+
+class DropDownComponent(
+    initialSelection: Int,
+    private val options: List<String>,
+    private val maxDisplayOptions: Int = 6,
+) : UIBlock() {
+
+    private val writableExpandedState: State<Boolean> = BasicState(false)
+    private val optionContainerHeight = 18
```
Comment (Johni0702) [inline src/main/kotlin/gg/essential/vigilance/gui/settings/DropDownComponent.kt:26]: This is different from what we are using in Essential (20). Why is that? If it's intentional, then should we maybe allow it to be configured via a constructor parameter?

**Review by Johni0702 — CHANGES_REQUESTED** [review]

File: src/main/kotlin/gg/essential/vigilance/gui/SettingsCategory.kt [hunk 2]
```diff
@@ -107,7 +107,7 @@ class SettingsCategory(category: Category) : UIContainer() {
 
         scroller.setVerticalScrollBarComponent(scrollBar, true)
 
-        scroller.onMouseScroll {
+        scroller.addScrollAdjustEvent(true) { _, _ ->
```
Comment (Johni0702) [inline src/main/kotlin/gg/essential/vigilance/gui/SettingsCategory.kt]: I guess practically it doesn't make a difference cause the ScrollComponent invokes both all the time, but logically it should be `isHorizontal = false`, no?

File: src/main/kotlin/gg/essential/vigilance/gui/SettingsCategory.kt [hunk 3]
```diff
@@ -107,7 +107,7 @@ class SettingsCategory(category: Category) : UIContainer() {
 
         scroller.setVerticalScrollBarComponent(scrollBar, true)
 
-        scroller.addScrollAdjustEvent(true) { _, _ ->
+        scroller.onMouseScroll {
```
Comment (Johni0702) [inline src/main/kotlin/gg/essential/vigilance/gui/SettingsCategory.kt:110]: You reverted this again. Why?

File: src/main/kotlin/gg/essential/vigilance/gui/settings/DropDownComponent.kt [hunk 4]
```diff
@@ -85,8 +92,8 @@ class DropDownComponent(
         x = 2.pixels(alignOpposite = true)
```
Comment (Johni0702) [inline src/main/kotlin/gg/essential/vigilance/gui/settings/DropDownComponent.kt:93]: Another case of padding that's now handled by `scrollerContainer`.
```suggestion
        x = 0.pixels(alignOpposite = true)
```

**Review by Johni0702 — CHANGES_REQUESTED** [review]

**Review by Johni0702 — APPROVED** [review]

**Comment by BlackBeltPanda** [comment]
Need to add one or two more things before merging this in.

File: src/main/kotlin/gg/essential/vigilance/gui/SettingsGui.kt [hunk 5]
```diff
@@ -166,53 +144,46 @@ class SettingsGui(
...
+                        child.component.incrementBy(.05f)
+                    }
+                    is NumberComponent -> if (keyCode == UKeyboard.KEY_UP) {
+                        child.component.increment()
+                    } else if (keyCode == UKeyboard.KEY_DOWN) {
+                        child.component.decrement()
+                    }
+                    is SwitchComponent -> when (keyCode) {
+                        UKeyboard.KEY_LEFT -> if (!child.component.enabled) child.component.click()
+                        UKeyboard.KEY_RIGHT -> if (child.component.enabled) child.component.click()
```
Comment (Johni0702) [inline src/main/kotlin/gg/essential/vigilance/gui/SettingsGui.kt:161]: Now that the black part is the knob, this needs to be swapped again.

**Review by Johni0702 — CHANGES_REQUESTED** [review]

**Review by Johni0702 — APPROVED** [review]

**Pull request merged**